### ISSUE TITLE
Withdraw function for expired transfer is lost when user loses his browser local storage

### ISSUE THREAD

**Issue by GabrielBuragev** [beamer-bridge/beamer#922]
The current transfer activity storage strategy is using the browser's `localStorage` as the only source.

This was proven to work so far on testnet where the conditions are near perfect, that is:
- Networks have little to no delay
- Active agents are properly funded and controlled internally
- Very low chances that a transfer gets expired

The mainnet situation is a little bit different, that is:
- Networks can go offline for hours, sometimes days
- Active agents might go offline due to insufficient funds, an unexpected crash or LP's simply removing their agents from our network. The probability for this case is lower since all of the agents will need to go offline for this scenario to apply. 
- Transaction delays might occur at any time for reasons unkown to us

With all of these mainnet cases, and probably even more, the probability that we will see cases where the user's transfers expire, is growing. ATM, the users depend on withdrawing their funds from their expired transfers via a `withdraw` button shown inside the transfer activity on the dapp once the conditions are met. Transfer activity is being stored in `localStorage`. 
The problem with the current storage strategy is that once a user loses his browser's `localStorage` (switching to another machine, reinstalling browser, reinstalling OS etc.) he also loses the ability to withdraw his funds from old expired transfers that haven't been withdrawn yet since the transfer activity history is gone.

**Comment by czepluch** [comment]
I think we should be able to handle this case somehow in the future. But for now, I would hope that it's possible to add a warning about this in the dApp.

I would assume that the bad case happens quite rarely as expiration is quite short and I wouldn't expect users to have expired transfers sitting idle for many days. That being said it would be nice to find a way to make sure that this cannot happen in the future. Maybe we can "just" query our subgraph and fetch expired, but unclaimed transfers. @fredo Do you have some input here?

**Comment by GabrielBuragev** [comment]
I agree that since we don't have any solution atm, we should at least inform the users about this possible inconvenience inside the disclaimer or elsewhere.

**Comment by fredo** [comment]
Thanks for this issue! I remember we discussed this problem in the past. The problem here is that fetching events for all history of the contract on behalf of the user will become quite expensive and worsens the UX due to long loading times.
That's why we went with the localStorage approach. 

There is still the problem, that the user relies on the local storage. And that is really independent from the occasions where requests do not get filled. In the future, as jacob mentioned we can leverage an indexing tool like the graph to query the user's transaction history

**Comment by GabrielBuragev** [comment]
As we are progressing with the adoption of graphs (https://github.com/beamer-bridge/beamer/issues/889) i prepared a solution proposal to this issue [here](https://github.com/beamer-bridge/beamer/issues/1125).
Lets use that issue to further track the progress of this.
